game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/RollValue.java

```diff
@@ -0,0 +1,31 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Value;
+
+/**
+ * Holds the value of a dice roll.
+ *
+ * <p>Handles infinite values and correctly limits the value
+ */
+@Value(staticConstructor = "of")
+@Getter(AccessLevel.NONE)
+class RollValue {
+
+  int value;
+  boolean isInfinite;
+
+  static RollValue of(final int value) {
+    return RollValue.of(value, value == -1);
+  }
+
+  RollValue add(final int extraValue) {
```

Comment:
It looks like we get a no-op when the roll value is infinite. Could that be a silent bug? Why would a caller want to invoke a method for a no-op, it seems that the caller did not realize it was dealing with an infinite roll value. 

I think this leads to some bigger questions. Perhaps `RollValue` is *not* just a simple integer and should not be exposed as that. We have a `getValue()` method as well that returns a magic number to indicate if the roll value is infinite (which is a SRP violation, the number is then context dependent and either it means the roll value, or if it is a magic number then it tell us the roll value is infinite). 

I think this is probably solved by handling the infinite case at the point of the caller. For example, looking at:
```
      return RollValue.of(unit.getUnitAttachment().getDefenseRolls(unit.getOwner()))
          .add(addSupport(unit, friendlySupportTracker))
          .add(addSupport(unit, enemySupportTracker))
          .getValue();
```

If the roll value is infinite:
- are we wasting support?
- is the roll value being returned a -1, or some other number?

It's not really clear for the above two questions.

Reply:
"-1" is how a map designer designates that a roll is infinite.  And if the map designer gives an AAroll bonus to a unit with infinite roll, than the bonus will be ignored since it doesn't make any sense.

And the caller shouldn't care if it is infinite or not infinite.  It is taking the map support rules as designed by the map designer and adding it to the roll.  Now, maybe it should throw an exception so that the map designer knows that they are trying to give an AAroll bonus to the unit but the game engine doesn't have a way to notify just the map designer.

As for passing around the RollValue instead of a magic number "-1", that will be a fairly large change since all the callers and usages are expecting ints.  It also shares the same int structures with non `RollValues`.  I'd rather do that as a separate PR.